### PR TITLE
docs: improve introductory documentation [MX-1073]

### PR DESCRIPTION
**Pull request by shamrt**
Completes [MX-1073](https://rewardops.atlassian.net/browse/MX-1073)

## Description

- Reorganizes, clarifies, completes most of the JSDoc markup, including: 
  - addition of missing `@async` tags and type definitions
  - addition of wholly missing JSDocs
- Also renames some private/protected functions, removing `*InContext` suffix
- Renames the documented `orders-recipients` module to `order-recipients`
- Changes the JSDoc template to [`better-docs`](https://www.npmjs.com/package/better-docs)
- Reorganizes README, and re-adds the Usage section with example methods

## Testing

- Run `npm run build:docs` 
- Peruse (bask in the wonder of) the output: `open docs/index.html`

### CODE REVIEW

**Review by Alex-Joseph — APPROVED** [review]
LGTM, HTML _on fleek_

**Review by Alex-Joseph — COMMENTED** [review]
Working on the failing test ATM 😅

File: README.md
Comment (shamrt) [reply README.md:44]: Yes… 👀

Comment (shamrt) [reply README.md:44]: Done! 🙏

File: README.md
Comment (shamrt) [reply README.md:105]: Is it required even for V4?

File: lib/resources/orders.js
Comment (shamrt) [reply lib/resources/orders.js:194]: Bold may not be a bad idea…

File: README.md
Comment (Alex-Joseph) [reply README.md:105]: Ah you are correct, it is **not** required

File: lib/resources/orders.js
Comment (shamrt) [reply lib/resources/orders.js]: Done

File: README.md
Comment (shamrt) [reply README.md:105]: Okay, we confirmed that for V4 it is _not_ required, so omitting for now.

**Comment by shamrt** [comment]
> Working on the failing test ATM 😅
> 
> UPDATE: Passing tests #37

While this PR is focused on docs, it has a few code touch-ups as well, so I'll wait until #37 is merged

**Pull request merged**
